software/mk/Makefile.builddefs

```diff
@@ -123,24 +124,29 @@ CRT_OBJ ?= crt.o
 
 ifeq ($(BSG_ELF_OFF_CHIP_MEM), 1)
   ifeq ($(BSG_ELF_DEFAULT_DATA_LOC), LOCAL)
-    LINK_SCRIPT ?= $(BSG_MANYCORE_DIR)/software/spmd/common/link_dmem.ld
+    LINK_GEN_OPTS ?= --default_data_loc=dmem --dram_size=$(BSG_ELF_DRAM_SIZE) --sp=$(BSG_ELF_STACK_PTR)
   else ifeq ($(BSG_ELF_DEFAULT_DATA_LOC), SHARED)
-    LINK_SCRIPT ?= $(BSG_MANYCORE_DIR)/software/spmd/common/link_dram.ld
+    LINK_GEN_OPTS ?= --default_data_loc=dram --dram_size=$(BSG_ELF_DRAM_SIZE) --sp=$(BSG_ELF_STACK_PTR)
   else
     $(error Invalid BSG_ELF_DEFAULT_DATA_LOC = $(BSG_ELF_DEFAULT_DATA_LOC); Only LOCAL and SHARED are valid)
   endif
 else ifeq ($(BSG_ELF_OFF_CHIP_MEM), 0)
   ifeq ($(BSG_ELF_DEFAULT_DATA_LOC), LOCAL)
-    LINK_SCRIPT ?= $(BSG_MANYCORE_DIR)/software/spmd/common/link_dmem2.ld
+    LINK_GEN_OPTS ?= --default_data_loc=dmem --dram_size=$(BSG_ELF_VCACHE_SIZE) --sp=$(BSG_ELF_STACK_PTR)
   else ifeq ($(BSG_ELF_DEFAULT_DATA_LOC), SHARED)
-    LINK_SCRIPT ?= $(BSG_MANYCORE_DIR)/software/spmd/common/link_dram2.ld
+    LINK_GEN_OPTS ?= --default_data_loc=dram --dram_size=$(BSG_ELF_VCACHE_SIZE) --sp=$(BSG_ELF_STACK_PTR)
   else
     $(error Invalid BSG_ELF_DEFAULT_DATA_LOC = $(BSG_ELF_DEFAULT_DATA_LOC); Only LOCAL and SHARED are valid)
   endif
 else
   $(error Invalid BSG_ELF_OFF_CHIP_MEM = $(BSG_ELF_OFF_CHIP_MEM); Only 0 and 1 are valid)
 endif
 
+LINK_SCRIPT ?= bsg_link.ld
+
+bsg_link.ld: $(LINK_GEN)
```

Comment:
I'd re-write this as `$(LINK_SCRIPT): $(LINK_GEN)` (and maybe depend on BSG_MACHINE_PATH/Makefile.machine.include?)

Reply:
Nope... we shouldn't generate if user provides a path for custom link script. May be I should put that in a comment.

Reply:
Hmm. Maybe there's a better way to specify it. What exactly are you trying to accomplish? Provide the ability to generate a link script in the directory that this makefile is included from?

Reply:
This target says, only generate `bsg_link.ld` and don't generate any other link script...

Reply:
Any suggestion to do this in a better way?